zds/forum/views.py

```diff
@@ -267,62 +299,74 @@ class TopicEdit(UpdateView, SingleObjectMixin, TopicEditMixin):
     @method_decorator(transaction.atomic)
     def dispatch(self, request, *args, **kwargs):
         self.object = self.get_object()
+
         if not self.object.forum.can_read(request.user):
             return redirect(reverse('cats-forums-list'))
+
         if ('text' in request.POST or request.method == 'GET') \
                 and self.object.author != request.user and not request.user.has_perm('forum.change_topic'):
             raise PermissionDenied
+
         if ('text' in request.POST or request.method == 'GET') \
                 and not self.object.first_post().is_visible and not request.user.has_perm('forum.change_topic'):
             raise PermissionDenied
+
         if 'page' in request.POST:
             try:
                 self.page = int(request.POST.get('page'))
             except (KeyError, ValueError, TypeError):
                 self.page = 1
+
         return super(TopicEdit, self).dispatch(request, *args, **kwargs)
 
     def get(self, request, *args, **kwargs):
         is_staff = request.user.has_perm('forum.change_topic')
+
         if self.object.author != request.user and is_staff:
             messages.warning(request, _(
                 u'Vous éditez ce sujet en tant que modérateur (auteur : {}). Soyez encore plus '
                 u'prudent lors de l\'édition de celui-ci !').format(self.object.author.username))
+
         form = self.create_form(self.form_class, **{
             'title': self.object.title,
             'subtitle': self.object.subtitle,
             'text': self.object.first_post().text,
             'tags': ', '.join([tag['title'] for tag in self.object.tags.values('title')]) or ''
         })
+
         return render(request, self.template_name, {'topic': self.object, 'form': form, 'is_staff': is_staff})
 
     def post(self, request, *args, **kwargs):
         if 'text' in request.POST:
             form = self.get_form(self.form_class)
 
-            if 'preview' in request.POST:
-                if request.is_ajax():
-                    content = render_to_response('misc/previsualization.part.html', {'text': request.POST['text']})
-                    return StreamingHttpResponse(content)
-                else:
-                    form = self.create_form(self.form_class, **{
-                        'title': request.POST.get('title'),
-                        'subtitle': request.POST.get('subtitle'),
-                        'text': request.POST.get('text'),
-                        'tags': request.POST.get('tags')
-                    })
-            elif form.is_valid():
+            if 'preview' not in request.POST and form.is_valid():
```

Comment:
Sauf erreur, tu as oublié le cas où ce n'est pas une prévisualisation (`'preview' not in request.POST`) et le formulaire n'est pas valide (`not form.is_valid()`) ! Ce qui, dans la version que tu as enlevé, donnait :

```python
form = self.get_form(self.form_class)

if 'preview' in request.POST: # False, on passe au suivant
    ...
elif form.is_valid(): # False, on passe au suivant
   ...

return render(request, self.template_name, {'topic': self.object, 'form': form})
```